eva_sub_cli/validators/validator.py

```diff
@@ -175,49 +177,82 @@ def _collect_validation_workflow_results(self):
         # Collect information from the output and summarise in the config
         if self.shallow_validation:
             self._collect_trim_down_metrics()
-        self._collect_vcf_check_results()
-        self._collect_assembly_check_results()
-        self._load_sample_check_results()
-        self._load_evidence_check_results()
-        self._load_fasta_check_results()
-        self._collect_metadata_results()
+
+        if VCF_CHECK in self.tasks:
+            self._collect_vcf_check_results()
+            self._load_evidence_check_results()
+
+        if ASSEMBLY_CHECK in self.tasks:
+            self._collect_assembly_check_results()
+            self._load_fasta_check_results()
+
+        if METADATA_CHECK in self.tasks:
+            self._collect_metadata_results()
+
+        if SAMPLE_CHECK in self.tasks:
+            self._load_sample_check_results()
 
     def _assess_validation_results(self):
         """
-            Assess if the validation results are meeting expectations and marks them as "PASS: true" or "PASS: false"
+            Assess if the validation results are meeting expectations and marks them as "PASS: true", "PASS: false"
+            or "PASS: Process not run yet" if the process as not been run yet.
```

Comment:
Personally I'd prefer we keep `pass` as a pure boolean value, just in case any downstream code checks `results['pass']` rather than `results['pass'] == True` (in which case a non-empty string is "truthy"). We could add another boolean for "did not run" or something similar.

(On the other hand I see that `verify_ready_for_submission_to_eva` does check the value is `True` which is probably the most important place... so, up to you I guess)

Reply:
updated